### PR TITLE
Fix #1392 added viewport into spatial filter list

### LINKED ISSUE #1392: Add viewport to the spatial filters list
Viewport, when selected, will get the current viewport bbox as a filter to perform search.

image

### PR DESCRIPTION
**Pull request by MV88**
The Viewport optios has been added to the spatial filter list.
image
Every time the view changes the geometry of the filterObj updates with an epic.

### CODE REVIEW

File: web/client/actions/queryform.js [hunk 1]
```diff
@@ -126,6 +128,18 @@ function selectSpatialMethod(method, fieldName) {
     };
 }
 
+function selectViewportSM() {
```
Comment (mbarto) [inline web/client/actions/queryform.js]: don't be shy, selectViewportSpatialMethod

File: web/client/components/data/query/QueryBuilder.jsx [hunk 2]
```diff
@@ -96,6 +96,7 @@ const QueryBuilder = React.createClass({
                 onChangeDrawingStatus: () => {},
                 onRemoveSpatialSelection: () => {},
                 onShowSpatialSelectionDetails: () => {},
+                onSelectViewportSM: () => {},
```
Comment (mbarto) [inline web/client/components/data/query/QueryBuilder.jsx]: don't be shy

File: web/client/actions/wfsquery.js [hunk 3]
```diff
@@ -103,12 +103,11 @@ function createQuery(searchUrl, filterObj) {
     };
 }
 
-function query(searchUrl, filterObj, retry) {
+function query(searchUrl, filterObj) {
```
Comment (mbarto) [inline web/client/actions/wfsquery.js]: is retry not used?

File: web/client/epics/wfsquery.js [hunk 4]
```diff
@@ -227,6 +228,33 @@ const closeFeatureEpic = action$ =>
...
+        extent[0] = -180.0;
+        extent[2] = 180.0;
+    }
+    return extent;
+}
+const viewportSelectedEpic = (action$, store) =>
+    action$.ofType(SELECT_VIEWPORT_SPATIAL_METHOD, CHANGE_MAP_VIEW)
+        .switchMap((action) => {
+            // calculate new geometry from map properties
+            const map = action.type === CHANGE_MAP_VIEW ? action : store.getState().map.present;
```
Comment (mbarto) [inline web/client/epics/wfsquery.js]: are you updating the geometry also if the spatial filter is not Viewport?

File: web/client/actions/wfsquery.js (hunk 3 above)
Comment (MV88) [inline web/client/actions/wfsquery.js]: nope, it's been removed in getFeatures method of DockerFeatureGrid component

**Pull request merged**